core/src/main/java/io/questdb/griffin/ExpressionParser.java

```diff
@@ -512,6 +512,17 @@ void parseExpr(GenericLexer lexer, ExpressionParserListener listener) throws Sql
                     case 'c':
                     case 'C':
                         if (SqlKeywords.isCastKeyword(tok)) {
+                            CharSequence caseTok = GenericLexer.immutableOf(tok);
+                            tok = SqlUtil.fetchNext(lexer);
+                            if (tok == null || tok.charAt(0) != '(') {
+                                lexer.backTo(position + 4, caseTok);
```

Comment:
I do appreciate our codebase is not ideal for magic numbers, but what is `4` ?

Reply:
At first glance it does look magical, but it's not that bad actually. `4` is the length  of the `case` keyword and allows me to backtrack if this is not a funtional call but reserved keyword usage for column name.

I thought this an was acceptable pattern for handling such cases because of the way [geohash](https://github.com/questdb/questdb/blob/8848438b8113725584f71f518e29064005ba4f0a/core/src/main/java/io/questdb/griffin/ExpressionParser.java#L333-L340) was handled. 

Maybe this would be more exemplary:

```
int CASE_KEYWORD_LENGTH = 4;
CharSequence caseTok = GenericLexer.immutableOf(tok);
tok = SqlUtil.fetchNext(lexer);
if (tok == null || tok.charAt(0) != '(') {
    lexer.backTo(position + CASE_KEYWORD_LENGTH, caseTok);
    tok = caseTok;
    processDefaultBranch = true;
    break;
}
```

Reply:
yep, this would be good

Reply:
Changed. Let me know if that works.